esp-hal/src/i2c/master/mod.rs

```diff
@@ -2188,6 +2324,20 @@ impl Driver<'_> {
     }
 }
 
+trait CheckTimeout {
```

Comment:
Any reason you prefer a trait over a plain old function?

Reply:
just a matter of taste - but no strong preference - it just looks "a bit" nicer at the call site IMHO

Reply:
I agree on the weird semantics